### PR TITLE
docs: update conflict detection documentation

### PR DESCRIPTION
**Pull request by multun**
- switch to a simpler requirement model
 - describe incremental requirement generation
 - describe the schema of spacing and routing requirements
 - rebrand the conflict diagram as a requirement diagram, show margins by default

Screenshot 2023-07-04 at 15-57-04 Conflict detection

### CODE REVIEW

File: content/docs/reference/design-docs/conflict-detection/index.en.md [hunk 1]
```diff
@@ -9,131 +9,210 @@ weight: 45
 This document is a work in progress
 {{% /pageinfo %}}
 
-Conflict detection is a process which enables figuring out if a timetable,
-comprised of many train paths, can be performed as planned.
+Conflict detection is the process of looking for timetable conflicts.
+A timetable conflict is any predictable condition which disrupts planned operations.
+Planned operations can be disrupted if a train slows down, is prevented from proceeding, or delayed.
```
Comment (eckter) [inline content/docs/reference/design-docs/conflict-detection/index.en.md:14]:
```suggestion
Planned operations can be disrupted if a train slows down, is prevented from proceeding, or is delayed.
```

**Review by Khoyo — APPROVED** [review]

File: content/docs/reference/design-docs/conflict-detection/index.en.md [hunk 2]
```diff
@@ -9,131 +9,210 @@ weight: 45
...
- - a resource does not have time to change state between non-overlapping, yet close, requirements
+In practice, the path of trains is partitioned into routes, which when set, ensure a train can safely follow the route.
 
-For example, there are **no** conflicts when:
- - a resource has non-overlapping exclusive requirements
- - a resource is shared by requirements which require the same configuration
+Routes have the following lifestyle:
 
-## Generating requirements
+ - As a train approaches the start of one of its routes, it is **called** by an operator. If all resources required to safely use the route are available, switches and level crossings start to move. If a resources is not available, say because another train has reserved a section of track, this process is delayed until all conditions are met.
```
Comment (eckter) [inline content/docs/reference/design-docs/conflict-detection/index.en.md:90]: Nitpicky, feel free to ignore this. "say" sounds too familiar in this context in my opinion

```suggestion
 - As a train approaches the start of one of its routes, it is **called** by an operator. If all resources required to safely use the route are available, switches and level crossings start to move. If a resources is not available, e.g. because another train has reserved a section of track, this process is delayed until all conditions are met.
```

File: content/docs/reference/design-docs/conflict-detection/index.en.md [hunk 3]
```diff
@@ -9,131 +9,210 @@ weight: 45
...
-These requirements are emitted as follows:
- - the route calling timeline is followed
- - every time the driver sees a signal:
-   - start by assuming routes reserved for routing should be reserved for spacing as well
-   - until the driver sees a signal which would cause a slowdown:
-     - attempt to make the last reserved zone occupied by a virtual train
-     - if the signal now causes a slowdown, the currently reserved zones are correct, continue to the next signal
-     - if not, unreserve the last zone of the path
+ - behind the tail of the next train
+ - at the end of the last route set for this train
```
Comment (eckter) [inline content/docs/reference/design-docs/conflict-detection/index.en.md:146]: Assuming the first option is for moving block, there's a missing case with the end of the last free block

**Review by eckter — COMMENTED** [review]
Nice

File: content/docs/reference/design-docs/conflict-detection/index.en.md (hunk 1 above)
Comment (multun) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: no need to repeat `is`. it parses as follows: `is (prevented from proceeding || delayed)`

File: content/docs/reference/design-docs/conflict-detection/index.en.md [hunk 4]
```diff
@@ -9,131 +9,210 @@ weight: 45
...
-## Conflict theory
+## Conflict modeling
 
-Conflict detection relies on an abstract model:
- - objects where conflicts happen are resources
- - resources are used by actors
- - resource uses can be incompatible
- - overlapping incompatible ressource uses generate a conflict
+**Actors** are objects which cause resources to be used:
+ - **train paths** (or someone / something on the behalf of the train)
```
[the remark this answers is not in the document]
Comment (multun) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:53]: train paths are scheduled trains. it's weird, I know

File: content/docs/reference/design-docs/conflict-detection/index.en.md (hunk 3 above)
Comment (multun) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:146]: there must be a misunderstanding: the movement authority always ends either at the end of the last set route, or somewhere behind a train which also uses the same set route.
what do you mean by free block?

Comment (eckter) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:146]: The misunderstanding was in the "somewhere behind a train" part. I understood this sentence as "the MA ends precisely at the tail of the next train".

But actually it is pretty clear, I just read it wrong.

File: content/docs/reference/design-docs/conflict-detection/index.en.md (hunk 1 above)
Comment (eckter) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: The sentence structure and comma placement feel wrong in that case, I believe `X if A, B, or C` is almost always interpreted as `(A | B | C) => X`

What about "Planned operations can be disrupted if a train slows down, i.e. when it is delayed or prevented from proceeding"?

Comment (Khoyo) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: "Planned operations can be disrupted if a train is slowed down, prevented from proceeding, or delayed."

If you don't want to repeat the is.

Comment (multun) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: > I believe X if A, B, or C is almost always interpreted as (A | B | C) => X

That's the intent

> What about "Planned operations can be disrupted if a train slows down, i.e. when it is delayed or prevented from proceeding"?

It does not have the same meaning:
 - slowing down is for when the train is moving, slows down, but does not stop
 - delayed for when a departure cannot occur at the planned time
 - prevented from proceeding line about when a route is not set in time, and the train has to stop

Comment (eckter) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: What about this version?

> Planned operations can be disrupted if a train **either** slows down, is prevented from proceeding, or delayed.

I asked for Garance's input, apparently the `is` isn't necessary like I though it was, but it still a bit hard to parse the initial sentence right (which could cause misunderstandings). Adding an `either` makes it clearer.

Comment (multun) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: either does not quite work, as it's usually used to express an alternative between *two* options. How do you feel about @Khoyo's version?

Comment (eckter) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: I do like his version

Comment (multun) [reply content/docs/reference/design-docs/conflict-detection/index.en.md:14]: Alright, let's roll with it then

Comment (multun) [inline content/docs/reference/design-docs/conflict-detection/index.en.md:14]:
```suggestion
Planned operations can be disrupted if a train is slowed down, prevented from proceeding, or delayed.
```

**Review by eckter — APPROVED** [review]
